orchestrator/careplanservice/handle_getquestionnaire.go

```diff
@@ -48,3 +50,79 @@ func (s *Service) handleReadQuestionnaire(ctx context.Context, request FHIRHandl
 		return []*fhir.BundleEntry{&bundleEntry}, []any{}, nil
 	}, nil
 }
+
+// handleSearchQuestionnaire performs a search for Questionnaire based on the user request parameters
+func (s *Service) handleSearchQuestionnaire(ctx context.Context, request FHIRHandlerRequest, tx *coolfhir.BundleBuilder) (FHIRHandlerResult, error) {
+	log.Ctx(ctx).Info().Msgf("Searching for Questionnaires")
+
+	bundle, err := s.searchQuestionnaire(ctx, request.QueryParams, request.FhirHeaders)
+	if err != nil {
+		return nil, err
+	}
+
+	results := []*fhir.BundleEntry{}
+
+	for _, entry := range bundle.Entry {
+		var currentQuestionnaire fhir.Questionnaire
+		if err := json.Unmarshal(entry.Resource, &currentQuestionnaire); err != nil {
+			log.Ctx(ctx).Error().
+				Err(err).
+				Msg("Failed to unmarshal resource for audit")
+			continue
+		}
+
+		// Create the query detail entity
+		queryEntity := fhir.AuditEventEntity{
+			Type: &fhir.Coding{
+				System:  to.Ptr("http://terminology.hl7.org/CodeSystem/audit-entity-type"),
+				Code:    to.Ptr("2"), // query parameters
+				Display: to.Ptr("Query Parameters"),
+			},
+			Detail: []fhir.AuditEventEntityDetail{},
+		}
+
+		// Add each query parameter as a detail
+		for param, values := range request.QueryParams {
+			queryEntity.Detail = append(queryEntity.Detail, fhir.AuditEventEntityDetail{
+				Type:        param, // parameter name as string
+				ValueString: to.Ptr(strings.Join(values, ",")),
+			})
+		}
+
+		bundleEntry := fhir.BundleEntry{
+			Resource: entry.Resource,
+			Response: &fhir.BundleEntryResponse{
+				Status: "200 OK",
+			},
+		}
+		results = append(results, &bundleEntry)
+
+		// Add audit event to the transaction
+		auditEvent := audit.Event(*request.LocalIdentity, fhir.AuditEventActionR, &fhir.Reference{
```

Comment:
if anyone can access the Questionnaire resource, do we need to audit it? @bramwesselo